test/uk/gov/hmrc/cgtpropertydisposals/actors/TaxEnrolmentRetryMongoActorSpec.scala

```diff
@@ -0,0 +1,151 @@
+import uk.gov.hmrc.http.HttpResponse
+
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
+class TaxEnrolmentRetryMongoActorSpec
+    extends TestKit(ActorSystem())
+    with ImplicitSender
+    with WordSpecLike
+    with Matchers
+    with BeforeAndAfterAll
+    with MockFactory {
+
+  override def afterAll: Unit =
+    TestKit.shutdownActorSystem(system)
+
+  val time: VirtualTime = new VirtualTime()
+
+  val mockTaxEnrolmentRepository = mock[TaxEnrolmentRetryRepository]
+
+  val maxStaggerTime = 1
+
+  val taxEnrolmentRetryMongoActor =
+    system.actorOf(TaxEnrolmentRetryMongoActor.props(time.scheduler, mockTaxEnrolmentRepository, maxStaggerTime))
+  val http4xxResponse = HttpResponse(403, None, Map.empty, None)
+  val http5xxResponse = HttpResponse(502, None, Map.empty, None)
+
+  def mockInsertTaxEnrolment(taxEnrolmentRequest: TaxEnrolmentRequest)(
+    response: Either[Error, Boolean]
+  ) =
+    (mockTaxEnrolmentRepository
+      .insert(_: TaxEnrolmentRequest))
+      .expects(taxEnrolmentRequest)
+      .returning(EitherT(Future.successful(response)))
+
+  def mockDeleteTaxEnrolment(userId: String)(
+    response: Either[Error, Int]
+  ) =
+    (mockTaxEnrolmentRepository
+      .delete(_: String))
+      .expects(userId)
+      .returning(EitherT(Future.successful(response)))
+
+  def mockUpdateStatusToFail(userId: String)(
+    response: Either[Error, Option[TaxEnrolmentRequest]]
+  ) =
+    (mockTaxEnrolmentRepository
+      .updateStatusToFail(_: String))
+      .expects(userId)
+      .returning(EitherT(Future.successful(response)))
+
+  def mockGetAllNonFailedEnrolmentRequests()(
+    response: Either[Error, List[TaxEnrolmentRequest]]
+  ) =
+    (mockTaxEnrolmentRepository.getAllNonFailedEnrolmentRequests _)
+      .expects()
+      .returning(EitherT(Future.successful(response)))
+
+  implicit val localDateArb: Arbitrary[LocalDateTime] = Arbitrary.apply[LocalDateTime](Gen.const(LocalDateTime.now()))
+  val taxEnrolmentRequest                             = sample[TaxEnrolmentRequest]
+
+  "A Tax Enrolment Mongo actor" must {
+
+    "not reply back with a message when there is an error recovering the tax enrolment requests from mongo" in {
+      mockGetAllNonFailedEnrolmentRequests()(Left(Error("Error recovering tax enrolment retry requests")))
+      taxEnrolmentRetryMongoActor ! GetAllNonFailedTaxEnrolmentRetryRequests
+      expectNoMessage()
+    }
+
+    //TODO: fix
+//    "reply back with a message when it has recovered tax enrolment requests from mongo" in {
+//      mockGetAllNonFailedEnrolmentRequests()(Right(List(taxEnrolmentRequest)))
+//      taxEnrolmentRetryMongoActor ! GetAllNonFailedTaxEnrolmentRetryRequests
+//      time.advance(FiniteDuration(2, TimeUnit.SECONDS))
+//      expectMsg(
+//        RetryTaxEnrolmentRequest(
+//          taxEnrolmentRequest.copy(numOfRetries = 0, timeToNextRetryInSeconds = 0, currentElapsedTimeInSeconds = 0)
+//        )
+//      )
+//    }
+
```

Comment:
I can''t get this to work. The test-actor should have received the RetryTaxEnrolmentRequest message but it doesn't actually receive anything.